agent/agent.c

```diff
@@ -19,24 +19,46 @@
 #include "config.h"
 
 #include "agent.h"
+#include <hwangsae/relay.h>
 
 struct _HwangsaeAgent
 {
   GApplication parent;
+
+  HwangsaeRelay *relay;
 };
 
 /* *INDENT-OFF* */
 G_DEFINE_TYPE (HwangsaeAgent, hwangsae_agent, G_TYPE_APPLICATION)
 /* *INDENT-ON* */
 
+static void
+hwangsae_agent_dispose (GObject * object)
+{
+  HwangsaeAgent *self = HWANGSAE_AGENT (object);
+
+  g_clear_object (&self->relay);
+
+  G_OBJECT_CLASS (hwangsae_agent_parent_class)->dispose (object);
+}
+
 static void
 hwangsae_agent_class_init (HwangsaeAgentClass * klass)
 {
+  GObjectClass *gobject_class = G_OBJECT_CLASS (klass);
+
+  gobject_class->dispose = hwangsae_agent_dispose;
 }
 
 static void
 hwangsae_agent_init (HwangsaeAgent * self)
 {
+  self->relay = hwangsae_relay_new ();
+
+  g_signal_connect_swapped (self->relay, "sink-connected",
+      (GCallback) g_application_hold, self);
```

Comment:
So the agent should run permanently? I thought it should stop when not in use because you created the agent's GApplication with `G_APPLICATION_IS_SERVICE` flag, which makes it exit after 10 seconds unless something is `g_application_hold()`ing it.

Reply:
Ah, yes, but in our concept, the relay agent runs as a daemon.

Reply:
```
G_APPLICATION_IS_SERVICE
Run as a service. In this mode, registration fails if the service is
already running, and the application will initially wait up to
10 seconds for an initial activation message to arrive.
```
So you're interested in that "registration" aspect of the flag and the timeout is just a side effect. In that case I can call `g_application_hold()` only once and keep the agent running permanently, right?

Reply:
Yes right. We don't have an activation message, and that's the way we did in _Gaeul_ agent.